### PR TITLE
Updated to use upring plugin system

### PR DESCRIPTION
**Pull request by delvedor**
As titled.

The `bin` component has been removed since is not useful anymore.
Also added *async-await* support.

### CODE REVIEW

**Review by lucamaraschi — APPROVED** [review]

**Comment by mcollina** [comment]
I think the `bin` component is still __very__ useful for testing and demos. Can you please restore it?

File: package.json [hunk 1]
```diff
@@ -23,28 +20,29 @@
...
-    "pre-commit": "^1.1.3",
-    "snazzy": "^5.0.0",
-    "standard": "^9.0.0",
-    "tap": "^10.0.0"
+    "coveralls": "^3.0.0",
+    "pre-commit": "^1.2.2",
+    "snazzy": "^7.0.0",
+    "standard": "^10.0.3",
+    "tap": "^10.7.3",
+    "upring": "^0.21.0"
```
Comment (mcollina) [inline package.json]: We have upring 0.22.0 now"

**Review by mcollina — CHANGES_REQUESTED** [review]
I think the `bin` component is still __very__ useful for testing and demos. Can you please restore it?

**Comment by delvedor** [comment]
@mcollina instead of restoring it, why not build a separate package just for this purpose?

**Comment by mcollina** [comment]
That’s also ok, can you add it to the org?

**Comment by delvedor** [comment]
Sure!
Let's merge and release this, then I'll work on the new package :)

**Comment by delvedor** [comment]
https://github.com/upringjs/kv-demo

**Review by mcollina — APPROVED** [review]
LGTM

**Pull request merged**